woodwork/statistics_utils/_get_dependence_dict.py

```diff
@@ -95,7 +100,19 @@ def _get_dependence_dict(
         valid_columns = pearson_columns
     if "mutual_info" in calc_order:
         mi_types = get_valid_mi_types()
-        mutual_columns = _get_valid_columns(dataframe, mi_types)
+        cols_to_drop = _find_large_categorical_columns(dataframe, max_nunique)
+        if len(cols_to_drop):
+            warnings.warn(
```

Comment:
We choose to give a warning rather than include these columns in the return in order to allow OS users to see things and to allow for minimal changes downstream. For our future purposes, we can filter the warnings and get the columns dropped, which we can then pass to users easily.

Reply:
I like it